src/components/Balance/History/FuturesHistoryTable.js

```diff
@@ -0,0 +1,166 @@
+
+/**
+ * @typedef {import("../../../store/initialState").DefaultState} DefaultStateType
+ * @typedef {import("../../../store/initialState").DefaultStateSession} StateSessionType
+ * @typedef {import("mui-datatables").MUIDataTableColumn} MUIDataTableColumn
+ * @typedef {import("mui-datatables").MUIDataTableMeta} MUIDataTableMeta
+ * @typedef {import("mui-datatables").MUIDataTableOptions} MUIDataTableOptions
+ * @typedef {import("../../../store/initialState").UserEquityEntity} UserEquityEntity
+ */
+
+/**
+ * Provides a table to display providers' profits stats.
+ *
+ * @typedef {Object} DefaultProps
+ * @property {string | React.ReactNode} title Table title.
+ * @property {'futuresDailyBalance'} persistKey Key to save display columns settings.
+ * @property {Array<UserEquityEntity>} list
+ * @property {Array<String>} quotes
+ *
+ * @param {DefaultProps} props Component props.
+ * @returns {JSX.Element} Component JSX.
+ */
+const FuturesHistoryTable = ({ title, persistKey, list, quotes }) => {
+  let data = [...list].sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime());
+
+  // /**
+  //  * Format Yes/No value.
+  //  * @param {Number} val Val.
+  //  * @returns {React.ReactNode} Formatted node.
+  //  */
+  // const renderPercentage = (val) => <span>{formatNumber(val, 2)}%</span>;
+
+  /**
+   * @type {Array<MUIDataTableColumn>} Table columns
+   */
+  let columns = [
+    {
+      name: "date",
+      label: "col.date",
+      options: {
+        display: "true",
+        viewColumns: true,
+      },
+    },
+    {
+      name: "pnlUSDT",
+      label: "col.dailyprofitlossUSDT",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+    {
+      name: "pnlBTC",
+      label: "col.dailyprofitlossBTC",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+    {
+      name: "sumPnlUSDT",
+      label: "col.cumprofitlossUSDT",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+    {
+      name: "sumPnlBTC",
+      label: "col.cumprofitlossBTC",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+    {
+      name: "netTransferUSDT",
+      label: "col.nettransferUSDT",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+    {
+      name: "netTransferBTC",
+      label: "col.nettransferBTC",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+    {
+      name: "totalWalletUSDT",
+      label: "col.walletbalanceUSDT",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+    {
+      name: "totalWalletBTC",
+      label: "col.walletbalanceBTC",
+      options: {
+        customBodyRender: formatFloat,
+      },
+    },
+  ];
+
+  /**
+   * @type {MUIDataTableOptions}
+   */
+  const options = {
+    sortOrder: { name: "date", direction: "desc" },
+  };
+
+  const dynamicColumns = () => {
+    for (let a = 0; a < quotes.length; a++) {
```

Comment:
Can we use `i` instead? ;)

Reply:
yes we can, this is just what I follow from school :D

Reply:
`i` is usually a coding standard.